### DISCUSSION TITLE
How to add users while filebrowser running in the container?

### DISCUSSION THREAD

**Discussion by Poulman** [filebrowser/filebrowser#1379] [Q&A] [answered]
Please, help me with some issues.
I want to create users by commands, while filebrowser is running in Docker. But when if running 
`docker exec -it filebrowser_container ./filebrowser users add user user`
when the container is up, I receive 
`2021/04/20 21:08:13 timeout`
Could you, please, help me with that problem?
Also will be great to see a sample of database.db with some configuration. Maybe, there are possibilities to create a user, rewriting this file. The autogenerated file is coded.
And one more question, how I can play with the frontend (I want to add additional info about files) in a docker container. I don't found source files in the container.

**Comment by ramiresviana** [comment, marked as answer]
Manipulating the database via CLI while Filebrowser is running is not possible. You can configure your container to run commands before starting the Filebrowser or use a existing database file with your configuration.

**Reply by Poulman** [reply]
Thank you for your response, Could you, please share a sample of the existing database file?

**Reply by ramiresviana** [reply]
That's very simple, you just need to download and run Filebrowser on your machine and configure it, a new `filebrowser.db` will be available in the same folder of the executable.

**Comment by tux456** [comment]
I think it's mostly related to the boltdb database lock limitation as discuted here: https://github.com/filebrowser/filebrowser/issues/627


We are also looking to any solution to add user while filebrowser is running... is look difficult w/o modification like: 
- In our specific use case we do not need a user setting access from web interface, we are fine with fixed settings. Does it make sense to have a special mode that disable (or write lock) completely the DB from the filebrowser process? 
- User auto-creation with a default scope (ex: /home/$USER ) option when we use a external authentication?
- API talking to filebrowser process to access config



Actually, the only alternative we find are:
- Pre-create a bunch of users ( user01... user99) and use nginx reverse proxy with basic auth
- Launch a new filebrowser instance for each new users.

Someone have better sugestion?

**Comment by tux456** [comment]
Seam to work by just copying the database in a temporary file ...  surely the best way to corrupt the database!!! 

```
function fb_update {
  COMMAND=$*
  FB=./filebrowser
  DB=./filebrowser.db
  TMPDB=$DB$(date  +%F-%T-%N)
  cp -a $DB $TMPDB
  $FB -d $TMPDB $COMMAND
  cp -a $TMPDB $DB
}
```

You can then run, for example:

`fb_update users add user4324 xxxxxxxx`

**Comment by mckaygerhard** [comment]
what about using a RDBMS? event a file database? or at leas change the database to sqlite format?
